plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/rename/RenameProcessorTests.java

```diff
@@ -74,4 +129,54 @@ public void testRenameExistingFieldNullValue() throws Exception {
         assertThat(ingestDocument.hasField(newFieldName), equalTo(true));
         assertThat(ingestDocument.getFieldValue(newFieldName, Object.class), nullValue());
     }
+
+    public void testRenameAtomicOperationSetFails() throws Exception {
+        Map<String, Object> document = new HashMap<String, Object>() {
+            private static final long serialVersionUID = 362498820763181265L;
+            @Override
+            public Object put(String key, Object value) {
+                if (key.equals("new_field")) {
+                    throw new UnsupportedOperationException();
+                }
+                return super.put(key, value);
+            }
+        };
+        document.put("list", Collections.singletonList("item"));
+
+        IngestDocument ingestDocument = RandomDocumentPicks.randomIngestDocument(random(), document);
+        Processor processor = new RenameProcessor(Collections.singletonMap("list", "new_field"));
+        try {
+            processor.execute(ingestDocument);
+            fail("processor execute should have failed");
+        } catch(UnsupportedOperationException e) {
```

Comment:
besides this scenario (wrapped Map), when would `hasField` return `true`  and `removeField` fail?